src/components/UserLogin/Login.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import "./Login.css";
+
+const Login = () => {
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [error, setError] = useState("");
+	const [emailError, setEmailError] = useState("");
+	const [showPassword, setShowPassword] = useState(false);
+	const navigate = useNavigate();
+
+	const validateEmail = (email) => {
+		const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+		return emailRegex.test(email);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		setError("");
+		setEmailError("");
+		if (!validateEmail(email)) {
+			setEmailError("Invalid email format");
+			return;
+		}
+
+		try {
+			const response = await fetch(
+				"http://localhost:8000/api/v1/auth/login",
+				{
+					method: "POST",
+					headers: {
+						"Content-Type": "application/json",
+					},
+					body: JSON.stringify({ email, password }),
+				},
+			);
+
+			if (response.ok) {
+				const data = await response.json();
+				localStorage.setItem("token", data.token);
```

Comment:
Doesn't have to be in this MR, but I think it would be great to set up a React context that holds this token and can call the login and register routes. That way, we can easily access the token from anywhere in our site by using the context.

This might be a good read for what I'm describing here: https://dev.to/miracool/how-to-manage-user-authentication-with-react-js-3ic5. I don't think we need to go crazy and have authenticated routes on our frontend, but the context itself would be really nice rather than loading from localstorage everywhere.